### DISCUSSION TITLE
CapacitorJS and Microfrontends

### DISCUSSION THREAD

**Discussion by VaughnKHHo** [ionic-team/capacitor#5744] [General] [answered]
Trying to explore microfrontend development with CapacitorJS, Angular and native mobile apps. Is it possible to develop a mobile app as a microfrontend host application and embed smaller separate microfrontend apps? I'd imagine that it would not be possible to load the embedded apps dynamically at runtime for the native mobile app. But if so, can it be done in a monorepo too? Have come across Ionic Portals but looking for a way to use only CapacitorJS and Angular.

**Comment by mlynch** [comment, marked as answer]
Yep! This is exactly what Portals is for and is our recommended solution for this problem https://ionic.io/portals
